packages/e2e-tests/specs/editor/various/adding-blocks.test.js

```diff
@@ -350,4 +350,16 @@ describe( 'adding blocks', () => {
 		await page.keyboard.type( 'Paragraph inside group' );
 		expect( await getEditedPostContent() ).toMatchSnapshot();
 	} );
+
+	// Check for regression of https://github.com/WordPress/gutenberg/issues/27586
+	it( 'closes the main inserter after inserting a single-use block, like the More block', async () => {
+		await insertBlock( 'More' );
+		await page.waitForSelector(
+			'.edit-post-header-toolbar__inserter-toggle:not(.is-pressed)'
+		);
+		const inserterPanels = await page.$$(
+			'.edit-post-layout__inserter-panel'
+		);
+		expect( inserterPanels.length ).toBe( 0 );
```

Comment:
Can you include an accessibility related check to verify that the inserted block has focus?

Reply:
Done 👍